monitoring/bin/deploy_monitoring_cluster.sh

```diff
@@ -65,10 +65,37 @@ fi
 # Elasticsearch Datasource for Grafana
 ELASTICSEARCH_DATASOURCE="${ELASTICSEARCH_DATASOURCE:-false}"
 if [ "$ELASTICSEARCH_DATASOURCE" == "true" ]; then
-  log_verbose "Provisioning Elasticsearch datasource for Grafana"
-  kubectl delete secret -n $MON_NS --ignore-not-found grafana-datasource-es
-  kubectl create secret generic -n $MON_NS grafana-datasource-es --from-file monitoring/grafana-datasource-es.yaml
-  kubectl label secret -n $MON_NS grafana-datasource-es grafana_datasource=1 sas.com/monitoring-base=kube-viya-monitoring
+
+  # Check to see if logging components have already been deployed.
+  if [[ $(kubectl get pods -A -l app.kubernetes.io/instance=v4m-fb -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) == 0 ]]; then
+    log_verbose "No logging components found.  Skipping Elasticsearch data source set up.";
+  elif [[ $(kubectl get pods -A -l app.kubernetes.io/instance=v4m-fb -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) > 1 ]]; then
+    log_verbose "Too many logging components found.  Skipping Elasticsearch data source set up.";
+  else
+    # Replace logging namespace name and password in grafana-datasource-es.yaml file
+    logNS=$(kubectl get pods -A -l app.kubernetes.io/instance=v4m-fb -o custom-columns=:metadata.namespace --no-headers | uniq)
+    monDir=$TMP_DIR/$MON_NS
+    mkdir -p $monDir
+    cp -R monitoring/grafana-datasource-es.yaml $monDir/grafana-datasource-es.yaml
+  
+    # Pull Elasticsearch password in temp directory file
+    adminPass="$(kubectl -n $logNS get secret internal-user-admin -o=jsonpath="{.data.password}" | base64 --decode)"
+    
+    # Replace placeholders
+    log_debug "Replacing logging namespace for files in [$monDir]"
+    if echo "$OSTYPE" | grep 'darwin' > /dev/null 2>&1; then
```

Comment:
I wonder if we should consolidate the OS check and dealing with OS-specific function calls earlier (in common.sh, maybe). Then we could use a single OS-agnostic custom function syntax throughout our code..with the custom function "translating" to generate the OS-specific calls.  Or, we can move to containerization and eliminate the need to worry about the OS variations. 😉